albumentations/augmentations/functional.py

```diff
@@ -1395,3 +1395,30 @@ def spatter(
         raise ValueError("Unsupported spatter mode: " + str(mode))
 
     return img * 255
+
+
+def split_uniform_grid(image_shape: Tuple[int, int], grid: Tuple[int, int]) -> np.ndarray:
+    """Splits an image shape into a uniform grid specified by the grid dimensions.
+
+    Args:
+        image_shape (Tuple[int, int]): The shape of the image as (height, width).
+        grid (Tuple[int, int]): The grid size as (rows, columns).
+
+    Returns:
+        np.ndarray: An array containing the tiles' coordinates in the format (start_y, start_x, end_y, end_x).
+    """
+    height, width = image_shape
+    n_rows, n_cols = grid
+
+    # Compute split points for the grid
+    height_splits = np.linspace(0, height, n_rows + 1, dtype=int)
+    width_splits = np.linspace(0, width, n_cols + 1, dtype=int)
+
+    # Initialize tiles coordinates
+    tiles = []
+
+    # Calculate tiles coordinates using list comprehension for efficiency
+    tiles = [(height_splits[i], width_splits[j], height_splits[i + 1], width_splits[j + 1]) 
+             for i in range(n_rows) for j in range(n_cols)]
```

Comment:
It looks like might be vectorized. Right now I can not figure out how to do this. But latter I will try show some workaround.

Reply:
I do not see how to do it either, and ChatGPT does not propose anything that works.

Also here `n_cols` and `n_rows` are rather small, less than 10 each, and most typical value is 2.

Hence vectorization would be beneficial, but benefits would be limited.